lib/spark.js

```diff
@@ -39,8 +41,34 @@ export default class Spark {
   }
 
   async executeRetrievalCheck (retrieval, stats) {
+    console.log(`Calling Filecoin JSON-RPC to get PeerId of miner ${retrieval.minerId}`)
+    try {
+      const peerId = await this.lookupMinerPeerId(retrieval.minerId)
+      console.log(`Found peer id: ${peerId}`)
+      stats.providerId = peerId
+    } catch (err) {
+      console.error(err)
+      // There are three common error cases:
+      //  1. We are offline
+      //  2. The JSON RPC provider is down
+      //  3. JSON RPC errors like when Miner ID is not a known actor
+      // There isn't much we can do in the first two cases. We can notify the user that we are not
+      // performing any jobs and wait until the problem is resolved.
+      // The third case should not happen unless we made a mistake, so we want to learn about it
+      if (err.name === 'FilecoinRpcError') {
+        // TODO: report the error to Sentry
+        console.error('The error printed above was not expected, please report it on GitHub:')
+        console.error('https://github.com/filecoin-station/spark/issues/new')
+      } else {
+        this.#activity.onError()
+      }
+      err.reported = true
+      // Abort the check, no measurement should be recorded
+      throw err
```

Comment:
I don't understand the need for this custom property `.reported` here.

If we remove the `console.error(err)` above, then we can let the outer catch handler log it. Also, the outer catch handler can call `#activity.onError()` itself.

I think the only custom error handling we need here is when `err.name === 'FilecoinRpcError'`, and in that case we can simply add a new log event. I'm proposing to refactor this catch handler like this:

```suggestion
      // There are three common error cases:
      //  1. We are offline
      //  2. The JSON RPC provider is down
      //  3. JSON RPC errors like when Miner ID is not a known actor
      // There isn't much we can do in the first two cases. We can notify the user that we are not
      // performing any jobs and wait until the problem is resolved.
      // The third case should not happen unless we made a mistake, so we want to learn about it
      if (err.name === 'FilecoinRpcError') {
        // TODO: report the error to Sentry
        console.error('Unexpected error, please report it on GitHub:')
        console.error('https://github.com/filecoin-station/spark/issues/new')
      }
      // Abort the check, no measurement should be recorded
      throw err
```

Reply:
Let's discuss.

In my proposed implementation:
- FilecoinRpcError does not trigger `this.#activity.onError()`, therefore Spark stays in "online" mode
- We print the call to report the issue after the error stack.

In your proposal:
- FilecoinRpcError triggers `this.#activity.onError()` and puts Spark to offline mode. 
- The call to report the issue is printed before the error to report. (I guess that's fine, we just need to tweak the message.)

Here is the important question we need to answer: How should we handle the case when we cannot get a miner's Peer ID because Filecoin RPC returns an error? 
- This can be either because we are calling the RPC API incorrectly or because the miner ID is not a valid one. (I suppose we could also receive this error if the RPC API server has an internal problem; it can process the JSON-RPC part, but the called RPC method fails.)
- Do we want to indicate offline mode? The next retrieval check will likely be able fetch the round details from spark-api, which means Spark goes back to online mode, only to go offline again when the JSON RPC call fails.

Reply:
Ah understood! I thought the purpose of `err.reported` was to prevent double log lines, now I see it's also to prevent reporting it to the activity state handler.

Could we be more explicit about this?
1. Rename to `error.setActivityToOffline = false`
2. Let the outer error handler handle logging, to be more consistent with other errors

Reply:
See db3b2ae